url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/cmdlib/LoadLibrary.java

```diff
@@ -0,0 +1,251 @@
+        lEngine.setMacroRepository(aEval.getMacroRepo());
+        
+        try
+        {
+            for(Loader lLoader: aLoaders)
+            {
+                loaders.remove(lLoader);
+
+                InputStream lIn = null;
+                try
+                {
+                    lIn = lLoader.getStream();
+                    lEngine.startNonInteractive(lIn);
+                }
+                catch (ReplEngineException e)
+                {
+                    final String lMsg = String.format("While loading '%s'.\n%s", lLoader.toString(), e.getMessage());
+                    throw new CommandException(lMsg);
+                }
+                finally
+                {
+                    // Cleanup. Ignore all exceptions here.
+                    if(lIn != null) try{lIn.close();}catch(Exception ignored){}
+                }
+                
+                loaders.add(lLoader);
+            }
+        }
+        
+        finally
+        {
+            aContext.setBinding(ReplEngine.INPUT, lInput);
+            aContext.setBinding(ReplEngine.OUTPUT, lOutput);
+            aContext.setBinding(ReplEngine.ERROR, lError);
+        }
+    }
+
+    public static interface Loader
+    {
+        InputStream getStream() throws CommandException;
+        void checkValidity() throws CommandException;
+    }
+
+    public static class FileLoader
+    implements Loader
+    {
+        private File file;
+
+        public FileLoader(File aFile)
+        {
+            file = aFile;
+        }
+
+        public FileLoader(String aPath)
+        {
+            file = new File(aPath);
+        }
+
+        public void checkValidity() throws CommandException
+        {
+            if(!file.exists() || !file.isFile())
+                throw new CommandException(String.format("The file does not exist: \"%s\".", file.getAbsolutePath()));
+            if(!file.canRead())
+                throw new CommandException(String.format("The file is not readable: \"%s\".", file.getAbsolutePath()));
+        }
+
+        public InputStream getStream()
+                throws CommandException
+        {
+            try
+            {
+                return new FileInputStream(file);
+            }
+            catch (FileNotFoundException e)
+            {
+                throw new CommandException(String.format("Exception while opening the file: '%s'.\n%s",file.getAbsolutePath(), e.getMessage()), e);
+            }
+        }
+
+        @Override
+        public String toString()
+        {
+            return file.getAbsolutePath();
+        }
+    }
+
+    public static class ClasspathLoader
+    implements Loader
+    {
+        String resource;
+
+        public ClasspathLoader(String aResource)
+        {
+            resource = aResource;
+        }
+
+        public void checkValidity() throws CommandException
+        {
+            if (this.getClass().getResourceAsStream(resource) == null)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *getResourceAsStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).